### DISCUSSION TITLE
AnalogWrite() PWM Attinyx06

### DISCUSSION THREAD

**Discussion by Rollpopptery** [SpenceKonde/megaTinyCore#694] [Q&A] [answered]
Is there an example using AnalogWrite(), or PWM  with megatinycore

I found the following does not work on ATtiny406. 

#define SPEED_PIN (13)

pinMode(SPEED_PIN, OUTPUT)
analogWrite(SPEED_PIN, 110)

result:  the pin ("PC3", IC-pin15, Digital 13), is low.

Thanks !

**Comment by Rollpopptery** [comment, marked as answer]
I see now that the 6 PWM pins are clearly marked with '~' in the diagram 'ATtiny_x06.gif'  .  I would like to remove my original question from the list, however the method by which to achieve this is not apparent.

**Comment by SpenceKonde** [comment]
This makes me wonder...,.. Should I have a tools submenu to select the PWM configuration? Maybe I should.....

Choose between PA3/4/5 and PC3/4/5 for the high half, or run PWM in non-split-mode with buffering..... Except neither of the two sets of pin options for WO0/1/2 (there's no 3/4/5 in non-split mode) are unarguably a great choice. because of the other functions of those pins.

**Reply by SpenceKonde** [reply]
It's PB0/1/2, PB3/4/5 for WO0/1/2 
PB0/1 are default SCL/SDA, PB2/3 are default serial or watch crystal. and PB4/5 don't exist on 14-pin parts,. You could get 2 channels back on 14-pin 1-series parts by using TCD0 on PA4/5 (at cost of a several hundred bytes of flash! TCD PWM implementation is bloated AF)

The main motivation to doing something like that IMO would be to be able to use buffered mode on TCA0 PWM.
